File: prompts/fixtures/go/debug_crash.go
package main

import (
	"fmt"
	"os"
	"unsafe"
)

func divide(a, b int) int {
	return a / b               // line 10 — will panic when b=0
}

func dereferenceNil() {
	var p *int                 // line 14
	fmt.Println(*p)            // line 15 — nil pointer dereference
}

func abortHandler() {
	panic("abort requested")  // line 19 — panic
}

func main() {
	if len(os.Args) < 2 {                              // line 23
		fmt.Printf("Usage: %s [divzero|nil|abort]\n", os.Args[0]) // line 24
		os.Exit(1)                                      // line 25
	}
	mode := os.Args[1]                                 // line 27
	fmt.Printf("mode: %s\n", mode)                    // line 28
	_ = unsafe.Pointer(nil)
	if mode[0] == 'd' {                               // line 30
		divide(10, 0)                                  // line 31
	} else if mode[0] == 'n' {                         // line 32
		dereferenceNil()                               // line 33
	} else if mode[0] == 'a' {                         // line 34
		abortHandler()                                 // line 35
	}
}

File: prompts/fixtures/rust/debug_crash.rs
use std::env;
use std::process;

fn divide(a: i32, b: i32) -> i32 {
    a / b                          // line 5 — will panic when b=0
}

fn dereference_null() {
    let p: *const i32 = std::ptr::null();  // line 9
    unsafe {
        println!("{}", *p);        // line 11 — SIGSEGV
    }
}

fn abort_handler() {
    process::abort();              // line 16 — SIGABRT
}

fn main() {
    let args: Vec<String> = env::args().collect();  // line 20
    if args.len() < 2 {                             // line 21
        println!("Usage: {} [divzero|null|abort]", args[0]);  // line 22
        process::exit(1);                           // line 23
    }
    let mode = &args[1];                            // line 25
    println!("mode: {}", mode);                    // line 26
    if mode.starts_with('d') {                     // line 27
        divide(10, 0);                              // line 28
    } else if mode.starts_with('n') {              // line 29
        dereference_null();                         // line 30
    } else if mode.starts_with('a') {              // line 31
        abort_handler();                            // line 32
    }
}
